src/components/results/ResultTables.js

```diff
@@ -0,0 +1,85 @@
+import {
+  CircularProgress,
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow
+} from "@mui/material";
+import {hooks} from "../../config/queryClient";
+import {useContext} from "react";
+import {QuizContext} from "../context/QuizContext";
+import {getAppDataByQuestionId, getDataWithId} from "../context/utilities";
+
+const ResultTables = () => {
+
+  const {data: responses, isLoading} = hooks.useAppData();
+  const {order, questions} = useContext(QuizContext);
+
+
+  if (isLoading) {
+    return <CircularProgress/>;
+  }
+
+
+  /**
+   * Helper function to get the name of all users that have answered to at least one question.
+   *
+   * @returns Immutable set of user's memberId
+   */
+  const getAllUser = () => {
+    return responses.map(r => r.memberId).toSet();
```

Comment:
You don't really need the `toSet` I think.

Reply:
Actually I used a `Set` here to automatically remove the duplicates in my list, as when I retrieve the `memberId` from  `responses` they will be duplicated.

I didn't find a better way to retrieve the `memberId` of users that answered the quiz than by filtering the `responses` list. Do you think there could be another way to directly get unique `memberId` ?